### ISSUE TITLE
Support for many workspaces

### ISSUE THREAD

**Issue by CarloCattano** [KZDKM/Hyprspace#21]
First of all thanks for this amazing tool ! 

It would be beneficial to have a way to manage having many workspaces displayed, or able to navigate them in more detail.
![20240415_12h12m25s_grim](https://github.com/KZDKM/Hyprspace/assets/17380530/886b1637-96aa-4e74-ba76-5d4dcd7e1367)

An option would be a Dinamic scaling of the workspace that is workspace count dependant, so the more workspaces the more tiny the previews get ?
or

Allowing to Drag somehow the hyprspace panel to _rotate_ the workspace set visible.

Say you want to drag an app from ws 1 to ws.length, and if not in view, is hard to do.

Another way would be to have a multi-line workspace bar , or an option to center it floating on the screen.

Example from my dirty hack workspacer
![image](https://github.com/KZDKM/Hyprspace/assets/17380530/203708a0-4ee2-438f-8659-b3d076db1b8c)

![image](https://github.com/KZDKM/Hyprspace/assets/17380530/6f3e8f09-790f-4b5a-9233-b5a8b4baaf2a)

I would love to give you a hand but I'm really unfamiliar with wlroots and such in cpp , but the code is well written and understandable

**Comment by KZDKM** [comment]
Thanks for the suggestion! Dynamic height based on the number of workspaces is planned, the configuration values for it are already in the code but I did not get to implement yet.

**Comment by KZDKM** [comment]
Reading this issue again in more detail, I think you did not realize that the plugin already supports navigating the workspace panel by scrolling the mousewheel or swiping with touchpad when your cursor is on the panel. Anyways, I should make this clearer in readme.

**Comment by CarloCattano** [comment]
Right , I couldn't test mouse-wheel but indeed a 3 finger track-pad swipe rotates it as expected.
Thanks again

**Comment by saipavanc** [comment]
Yes, this feature is also useful for multi monitor setup, when the height of the workspace panel cannot just be set with pixel size.

**Comment by jficz** [comment]
> useful for multi monitor setup

exactly, if the height could be set in `%` as well, that would be a huge benefit for multimonitor.
